src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -332,19 +440,39 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
 
+        // Make sure journey state indicates that the trip should still possible on at least one day of the week as of
+        // the last time a matching itinerary was attempted to be found. This field in the journey state is used instead
+        // of the trip's itinerary existence data in order to not skip trips that are monitored only one day of the week
+        // and may have not had a matching itinerary on that single monitored day even though it could be possible the
+        // next week.
+        if (previousJourneyState.noLongerPossible) return true;
+
         // get the most recent journey state itinerary to see when the next monitored trip is supposed to occur
-        Itinerary latestMatchingItinerary = trip.journeyState.matchingItinerary;
-        boolean matchingItineraryActiveOrUpcoming = latestMatchingItinerary != null &&
-            latestMatchingItinerary.endTime.after(new Date(trip.journeyState.lastCheckedEpochMillis));
+        boolean matchingItineraryActiveOrUpcoming = previousMatchingItinerary != null &&
+            previousMatchingItinerary.endTime.after(new Date(previousJourneyState.lastCheckedEpochMillis));
         if (matchingItineraryActiveOrUpcoming) {
-            matchingItinerary = latestMatchingItinerary;
-            targetDate = trip.journeyState.targetDate;
+            // skip checking the trip the rest of the time that it is active if it had its trip status marked as the
+            // next trip not being possible
+            if (previousJourneyState.tripStatus == TripStatus.NEXT_TRIP_NOT_POSSIBLE) return true;
+
+            matchingItinerary = previousMatchingItinerary;
+            targetDate = previousJourneyState.targetDate;
         } else {
             // Either the monitored trip hasn't ever checked on the next itinerary, or the most recent itinerary has
             // completed and the next possible one needs to be fetched in order to determine the scheduled start time of
             // the itinerary on the next possible day the monitored trip happens
 
             LOG.info("Calculating next itinerary for trip");
+
+            // initialize the trip's journey state and matching itinerary to the latest journeyState's matching
+            // itinerary, or use the itinerary that the trip was saved with
+            if (previousMatchingItinerary == null) {
+                // clone the trip's itinerary just in case the code attempts to save the trip (and thus the itinerary)
+                matchingItinerary = trip.itinerary.clone();
+            } else {
+                matchingItinerary = previousMatchingItinerary.clone();
+            }
+
             // calculate target time for the next trip plan request
             // find the next possible day the trip is active by initializing the the appropriate target time. Start by
             // checking today's date at the earliest in case the user has paused trip monitoring for a while
```

Comment:
On line 479: opportunity to reuse: 
```java
targetZonedDateTime = trip.tripZonedDateTime(DateTimeUtils.nowAsLocalDate());
```
?

Reply:
Updated in 1987213b6754ffa98f4748820b628be64f9443f4